main/FatesInventoryInitMod.F90

```diff
@@ -374,6 +377,82 @@ subroutine initialize_sites_by_inventory(nsites,sites,bc_in)
 
          deallocate(patch_pointer_vec,patch_name_vec)
 
+         ! now that we've read in the patch and cohort info, check to see if there is any real age info
+         if ( (sites(s)%youngest_patch%age .eq. sites(s)%oldest_patch%age) .and. &
```

Comment:
This is just a feeling I have, but I never like testing equivalence of real numbers.  I know in this case they are all read in as the same number, but I wonder if certain compilers or optimizations may mess up the last digit or something, and thus this equivalence would not work even if they numbers are the same in file.

I like comparing their absolute difference to nearzero. 

@glemieux @ckoven  am I just being paranoid here?

Reply:
@rgknox just because you're paranoid doesn't mean they aren't after you.

Reply:
@rgknox I think that makes sense.

Reply:
By the way, I do remember being taught in Numerical Analysis to do just what you suggest and NOT test for exact values but for within a tolerance, or using inequalities. So you are recommending a best practice. And experience definitely shows that floating point arithmetic really is out to get you!

Reply:
@ekluzek I'm taking that exact class right now!

Reply:
who said that?   I know someone is there ...